### PR TITLE
hf_nml_outbnd: Minor bug in ww3_grid.ftn: Info about outbound files

### PR DESCRIPTION
**Pull request by CarstenHansen**
In the output from ww3_grid, some info text is missing about the last outbound boundary file for one-way nesting: "File nestI.ww3 Number of points .."  when namelists input files are used.

Solution: Let NPO=0 mark that we have got the info for the last output b.c. file. Then the do-loop over outbound lines will exit immediately after the block that writes the info.

Additional correction: The file ww3_grid.scratch isn't used when namelists input files are used. Therefore, also prepend a condition "IF ( .NOT. FLGNML )" to other statements handling the file NDSS.

### CODE REVIEW

**Comment by ajhenrique** [comment]
Thanks @CarstenHansen. @mickaelaccensi please have a look at the proposed fix in regards to namelists and comment.

File: model/ftn/ww3_grid.ftn [hunk 1]
```diff
@@ -4876,7 +4876,9 @@
 !     ILOOP = 2 to fill data arrays
 !
       WRITE (NDSO,990)
-      OPEN (NDSS,FILE=TRIM(FNMPRE)//'ww3_grid.scratch',FORM='FORMATTED')
+! ww3_grid.scratch is not used with namelist input (C.Hansen 20190701)
```
Comment (mickaelaccensi) [inline model/ftn/ww3_grid.ftn]: to remove and mention in commit message

File: model/ftn/ww3_grid.ftn [hunk 2]
```diff
@@ -4906,21 +4910,29 @@
...
               I=I+1
             ELSE
-              EXIT
+!              EXIT
+! Bug (minor): If we exit here, a code block below will not be reached that
+! writes info about the last boundary file:'File nestI.ww3  Number of points ..'
+! ( The block does WRITE(NDSO,991), WRITE (NDSO,992), etc.)
+! Instead, let NPO=0 mark that we have got the info for the last file, and the
+! do-loop over outbound lines will exit immediately after the block. This is a
+! minor bug because it only affects output messages (C. Hansen 20190701)
```
Comment (mickaelaccensi) [inline model/ftn/ww3_grid.ftn]: to remove and mention in the commit message

File: model/ftn/ww3_grid.ftn [hunk 3]
```diff
@@ -4906,21 +4910,29 @@
...
 !
-          IF ( ILOOP .EQ. 1 ) THEN
-            IF (FLGNML) THEN
-              WRITE(NDSS,'(A)') NML_OUTBND_LINE(I-1)
-            ELSE
+          IF ( .NOT. FLGNML .AND. ILOOP .EQ. 1 ) THEN
+!            IF (FLGNML) THEN
+! NML_OUTBND_LINE are not strings (C. Hansen 20190701), but NDSS isn't used
+!              WRITE(NDSS,'(A)') NML_OUTBND_LINE(I-1)
+!            ELSE
```
Comment (mickaelaccensi) [inline model/ftn/ww3_grid.ftn]: to remove and mention in the commit message

**Review by mickaelaccensi — CHANGES_REQUESTED** [review]
I agree with the bugfix correction with NPO=0 instead of exit statement.
Carsten, can you remove all the your comments and commented lines to clean up the code before the pull request. All explanations of this bugfix must be tracked in the commit message only.

thanks for your help,

**Comment by ajhenrique** [comment]
@CarstenHansen can you please weigh in @mickaelaccensi comments above? I would also like to ask that you include the changes proposed by @MicahDavid with your remarks as per PR #57 bundled with your code changes. Thanks!

File: model/ftn/ww3_grid.ftn (hunk 3 above)
Comment (CarstenHansen) [reply model/ftn/ww3_grid.ftn]: I understand, thank you. I will do as you ask.

**Comment by CarstenHansen** [comment]
I will include the changes proposed by @MicahDavid and my suggested extension.

**Comment by mickaelaccensi** [comment]
bugfix integrated in PR #86 and referenced in issue #89

**Comment by CarstenHansen** [comment]
I browsed the file changes in develop #86. It completes this pull request, and also the changes proposed by @MicahDavid and my suggested extension. May I close this pull request?

**Comment by ajhenrique** [comment]
Embedded into Ifremer PR. Closing.

**Pull request closed without merging**